data/datagen.py

```diff
@@ -128,11 +128,14 @@ def _get_cl_type(self, i, j):
                 return 'cl_00'
             else:
                 return 'cl_0e'
-        else:
-            if (j < self.n_cl) or (i == self.n_tot - 1):
+        elif i < self.n_cl + self.n_sh:
+            if (j < self.n_cl) or (j == self.n_tot - 1):
                 return 'cl_0e'
             else:
                 return 'cl_ee'
+        else:
+            # Else = cmbk
```

Comment:
don't you need to check j here?

Reply:
I've added it just in case but it returned the correct data type since it was called always so that j >= i.

Reply:
yeah, just in case we change this in the future. Thanks!

Reply:
if this becomes more complicated I would rather update sacc and use `s.get_data_types(tracers=(tr1, tr2))`